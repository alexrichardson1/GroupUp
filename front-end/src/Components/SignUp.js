import React, { Component } from "react";
import { LinkContainer } from "react-router-bootstrap";

class SignUp extends Component {
  render() {
    return (
      <form>
        <h2>Create an Account!</h2>
        <fieldset>
          <ul>
            <li>
              <label for="username">Username:</label>
              <input type="text" id="username" required />
            </li>
            <li>
              <label for="email">Email:</label>
              <input type="email" id="email" required />
            </li>
            <li>
              <label for="password">Password:</label>
              <input type="password" id="password" required />
            </li>
          </ul>
        </fieldset>
        <LinkContainer to="/home">
          <button>Submit</button>
        </LinkContainer>
        <LinkContainer to="/">
          <button type="button">Have an Account?</button>
        </LinkContainer>
      </form>
    );
  }
}

export default SignUp;
