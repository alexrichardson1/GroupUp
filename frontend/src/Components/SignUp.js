import React, { Component } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { Form, Button } from "react-bootstrap";

class SignUp extends Component {
  render() {
    return (
      <div>
        <h2>Create an Account!</h2>
        <Form>
          <Form.Group controlId="formBasicEmail">
            <Form.Label>Email address</Form.Label>
            <Form.Control type="email" placeholder="Enter email" />
            <Form.Text className="text-muted">
              We'll never share your email with anyone else.
            </Form.Text>
          </Form.Group>
          <Form.Group controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <Form.Control type="password" placeholder="Password" />
          </Form.Group>
          <LinkContainer to="/">
            <Button>Login</Button>
          </LinkContainer>
          <LinkContainer to="/signup">
            <Button type="button">Create an Account</Button>
          </LinkContainer>
          <LinkContainer to="/forgotpassword">
            <Button>Forgot Password?</Button>
          </LinkContainer>
        </Form>
      </div>
      //     </ul>
      //   </fieldset>
      //   <LinkContainer to="/home">
      //     <button>Submit</button>
      //   </LinkContainer>
      //   <LinkContainer to="/">
      //     <button type="button">Have an Account?</button>
      //   </LinkContainer>
      // </form>
    );
  }
}

export default SignUp;
