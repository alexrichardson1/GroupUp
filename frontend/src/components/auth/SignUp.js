import React, { Component } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { Form, Button } from "react-bootstrap";
import NavBar from "components/NavBar";

class SignUp extends Component {
  componentDidMount() {
    document.title = "Sign Up";
  }

  render() {
    return (
      <div>
        <NavBar renderBool={[false, false, false, false]} loginPage={1} />
        <h2>Create an Account!</h2>
        <Form>
          <Form.Group>
            <Form.Label>Name</Form.Label>
            <Form.Control type="text" placeholder="Enter name" />
            <Form.Label>Email address</Form.Label>
            <Form.Control type="email" placeholder="Enter email" />
            <Form.Text className="text-muted">
              We'll never share your email with anyone else.
            </Form.Text>
          </Form.Group>
          <Form.Group>
            <Form.Label>Password</Form.Label>
            <Form.Control type="password" placeholder="Password" />
          </Form.Group>
          <LinkContainer to="/home">
            <Button
              onClick={() => {
                alert("Successfully created an account.");
              }}
            >
              Submit
            </Button>
          </LinkContainer>
          <LinkContainer to="/" activeClassName="">
            <Button>Already Have an Account?</Button>
          </LinkContainer>
        </Form>
      </div>
    );
  }
}

export default SignUp;
