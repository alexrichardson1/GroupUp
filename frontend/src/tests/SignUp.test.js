import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SignUp from "components/auth/SignUp";

test("renders correctly", async () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SignUp />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
