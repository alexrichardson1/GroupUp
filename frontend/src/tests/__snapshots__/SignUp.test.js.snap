// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navBar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <img
        alt="group up logo"
        className="logoImage"
        href="/home"
        onClick={[Function]}
        src="https://i.imgur.com/w92t1xH.png"
      />
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr=auto navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
          Sign Up
        </span>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="ml-auto navHelp navbar-brand"
          href="/help"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          title="Help"
        >
          <img
            alt="help"
            className="splitterImage"
            src="/info-circle.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <h2>
    Create an Account!
  </h2>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Name
      </label>
      <input
        className="form-control"
        placeholder="Enter name"
        type="text"
      />
      <label
        className="form-label"
      >
        Email address
      </label>
      <input
        className="form-control"
        placeholder="Enter email"
        type="email"
      />
      <small
        className="text-muted form-text"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Password
      </label>
      <input
        className="form-control"
        placeholder="Password"
        type="password"
      />
    </div>
    <a
      aria-disabled={true}
      className="btn btn-primary disabled"
      href="/home"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
    >
      Submit
    </a>
    <a
      className="btn btn-primary"
      href="/"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={Object {}}
    >
      Already Have an Account?
    </a>
  </form>
</div>
`;
